.circleci/config.yml

```diff
@@ -76,41 +78,72 @@ jobs:
                 RUSTFLAGS="--cfg procmacro2_semver_exempt" cargo install cargo-tarpaulin  --version ${TARPAULIN_VERSION} --force
             fi
       - restore_cache:
-          name: Restoring ./target Cache
+          # Actually master does not need to download the cache. However we cannot make conditions here.
+          name: Downloading ./target Cache from master
           keys:
-            - v2-target-{{ .Branch }}-{{ .Environment.CIRCLE_SHA1 }}
-            - v2-target-{{ .Branch }}
-            - v2-target
+            - v3-target-master
+      - run:
+          name: Restoring ./target Cache (non-master branch only)
+          command: |
+            if [[ "${CIRCLE_BRANCH}" == "master" ]]; then
+              echo "Current branch is $CIRCLE_BRANCH, skip"
+            else
+              echo "Current branch is $CIRCLE_BRANCH, restoring tar..."
+              tar xf ~/tikv_cache_build/target.tar || true
+            fi
+          no_output_timeout: 1800s
       - run:
           name: Building
           command: env SKIP_TESTS=true make trace_test
           no_output_timeout: 1800s
+      - run:
+          name: Caching ~/.cargo and ~/.local (master branch only)
+          command: |
+            rm -rf ~/tikv_cache_cargo
+            mkdir ~/tikv_cache_cargo
+            if [[ "${CIRCLE_BRANCH}" == "master" ]]; then
+              echo "Current branch is $CIRCLE_BRANCH, creating tar for ~/.cargo and ~/.local"
+              cd ~ && tar cf ~/tikv_cache_cargo/cargo.tar .cargo .local && cd -
+            else
+              echo "Current branch is $CIRCLE_BRANCH, skip"
+            fi
+          no_output_timeout: 1800s
       - save_cache:
-          name: Saving ~/.cargo Cache
-          key: v2-cargo-{{ .Branch }}-{{ checksum "Cargo.lock" }}
-          paths:
-            - ~/.cargo
-      - save_cache:
-          name: Saving ~/.local Cache
-          key: v2-local-{{ .Branch }}-{{ .Environment.CIRCLE_SHA1 }}
+          # This will be triggered for non-master as well. However the cache content is empty so it doesn't matter.
+          name: Uploading ~/.cargo and ~/.local Cache (master branch only)
+          key: v3-cargo-{{ .Branch }}-{{ .Environment.CIRCLE_SHA1 }}
           paths:
-            - ~/.local
-            - ~/.rustup
+            - ~/tikv_cache_cargo
       - run:
           name: Testing
           command: make trace_test
           no_output_timeout: 1800s
       - run:
-          name: Calculating code coverage
+          name: Calculating code coverage (master branch only)
```

Comment:
I guess this means no more reports per PR.

Reply:
yes. Once tarpaulin's bug is solved we can add code coverage back. Currently we are not very care about the code coverage in fact so temporarily removed.